### ISSUE TITLE
[no-use-before-define] no-use-before-define rule reports an wrong error for class method

### ISSUE THREAD

**Issue by Quramy** [typescript-eslint/typescript-eslint#2965]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-use-before-define": ["error"]
  }
}
```

```TS
// your repro code case

export class Hoge {
  foo({ a } = { a: 0 }) {
    return a;
  }
}
```

**Expected Result**

The `no-use-before-define` rule should not report error. The above example code is valid.

**Actual Result**

"'a' was used before it was defined. " error was reported

**Additional Info**

This issue occurs when I tried to update `@typescript-eslint/eslint-plugin` to v4.14.0 from v4.13.0 (The original PR is https://github.com/Quramy/ts-graphql-plugin/pull/325 )

And the following code, which has the similar argument definition, is still valid.

```ts
function foo({ a } = { a: 0 }) {
  return a;
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.0` |
| `@typescript-eslint/parser`        | `4.14.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.18.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
Same as #2941, #2942, #2945, #2947, #2950, #2957, #2961
